pkg/kapp/app/recorded_app_changes.go

```diff
@@ -122,3 +125,10 @@ func (a RecordedAppChanges) Begin(meta ChangeMeta) (*ChangeImpl, error) {
 
 	return change, nil
 }
+
+func (a RecordedAppChanges) changeLabelValue() string {
```

Comment:
anything we are doing to deal with renames of an app? (or is this future work?)

Reply:
As of now, we abandon all the app-changes in case of a rename (with user confirmation), so this check should be good for now to resolve the issue. The only difference that will be there now is that we will not be able to list the app-changes with old names. Example:
```
$ kapp deploy -a very-long-name -f config.yml

$ kapp rename -a very-long-name --new-name short-name

$ kapp app-changes list -a very-long-name
kapp: Error: App 'very-long-name' (namespace: name-limit) does not exist:
```
The error would happen because of the `meta` calls.

For future, we can completely migrate to using the label.